### ISSUE TITLE
Could this be implemented for R as well?

### ISSUE THREAD

**Issue by matthewacarson** [bitagoras/PyPadPlusPlus#12]
Hi. This isn't technically an issue, but I hope it's ok to ask anyway. I've been using Py Pad++ for a while, and I think it's awesome. I actually like it better than a lot of the IDEs because it's so lightweight, and you can easily run code line-by-line.

I'm also an R user, and I think R Studio is great and prefer it for full-blown projects, but it is not the most lightweight solution for small tasks where you're just writing 20 or so lines. Do you think it would be possible for someone to make an R plug in that functions like Py Pad++ ? I'm certainly not asking you to do it, but I'm just wondering if you have any insight about whether that would be possible. If yes, maybe this could be a project I work on in the future.

Thanks for all your hard work on this!

**Comment by bitagoras** [comment]
I think that there might be some difficulties to use PyPadPlusPlus as a basis for an R enhancement and probably it is more easy to write in C a specific R plugin for Notepad++ (but I have no experience with Npp plugin programming).

PyPadPlusPlus is fully written in Python and is using the PythonScript-Plugin, which can controll the user interface of Notepad++ (drawing the color border lines etc.). When a code line is executed with PyPadPlusPlus, there are two separate parts that have to work together:

- Inside the Notepad++ process some Python script is started for selecting the code lines and handling the data exchange between the interpreter and the UI of Npp.
- The interpreter subprocess on the other hand has not only to receive the lines and execute it. It has also to do the communication in a parallel thread and translate data between binary and text. And it has to generate the call tipps.

The first part, the Npp process, could be adapted to the R syntax and the interpreter interface. But the receiver side, which is another Python code (pyPad/pyPadClient.py), must be entirely implemented in R. And whether this is possible or not depends on the R language whether it supports e.g. threads and an appropriate binary to text transformation for the pipe. I never used R and don't know.

In summary I would say that the adaption of PyPadPlusPlus to R is probably too difficult.
